core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -350,16 +350,37 @@ int cuda_get_opt_block_size(const typename DriverType::functor_type& f,
                             const size_t vector_length,
                             const size_t shmem_extra_block,
                             const size_t shmem_extra_thread) {
-  return CudaGetOptBlockSize<
-      DriverType, LaunchBounds,
-      // LaunchBounds::launch_mechanism == Kokkos::Experimental::LaunchDefault ?
-      //            (( CudaTraits::ConstantMemoryUseThreshold <
-      //            sizeof(DriverType) )?
-      //                   Kokkos::Experimental::CudaLaunchConstantMemory:Kokkos::Experimental::CudaLaunchLocalMemory):
-      //             LaunchBounds::launch_mechanism
-      (CudaTraits::ConstantMemoryUseThreshold <
-       sizeof(DriverType))>::get_block_size(f, vector_length, shmem_extra_block,
-                                            shmem_extra_thread);
+
+    int kokkos_suggested_block_size;
+    /** TODO DZP:
+     * discuss if we want to give tools default values for their tuning parameters based on our calculations
+     * 1) Always
+     * 2) Never
+     * 3) As an option
+     *
+     * I've implemented 3, with the default option being to give the tool valid values. It's cumbersome.
+     */
+#ifndef KOKKOS_EXCLUSIVELY_USE_TUNED_VALUES 
+    kokkos_suggested_block_size = CudaGetOptBlockSize<
+        DriverType, LaunchBounds,
+        // LaunchBounds::launch_mechanism == Kokkos::Experimental::LaunchDefault ?
+        //            (( CudaTraits::ConstantMemoryUseThreshold <
+        //            sizeof(DriverType) )?
+        //                   Kokkos::Experimental::CudaLaunchConstantMemory:Kokkos::Experimental::CudaLaunchLocalMemory):
+        //             LaunchBounds::launch_mechanism
+        (CudaTraits::ConstantMemoryUseThreshold <
+         sizeof(DriverType))>::get_block_size(f, vector_length, shmem_extra_block,
+                                              shmem_extra_thread);
+#endif
+#ifdef KOKKOS_ENABLE_TUNING
+    if(Kokkos::Tuning::haveTuningTool()) {
+      size_t uniqIds[] = {0};
```

Comment:
Basically any [Context/Tuning] variable needs a unique ID. This is 0 because it isn't yet Kokkos::Tuning::giveMeAUnique[Tuning/Context]ID like it will be in the final version, and I started numbering them at `0`